journal/history.go

```diff
@@ -0,0 +1,63 @@
+package journal
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+var mux = sync.Mutex{}
+
+// Entry contains the command-line used to invoke the process associated with this item
+// and its journal ID.
+type Entry struct {
+	ID, Description string
+	Start           time.Time
+}
+
+// A list of commands that have been invoked
+var history = make([]Entry, 0)
+
+// Lookup to identify whether a journal ID has already been recorded in history. There's a small
+// chance the journal ID may not be unique in history. We handle that by ignoring any entries we
```

Comment:
When would it not be unique?

Reply:
OS's use a limited number for process IDs (I think 100k on macOS). If you somehow got the same process ID for the same executable, you'd have the same journal ID. I just realized I could include create_time (from https://github.com/shirou/gopsutil) in the journal ID to make it unique. Using `time.Now()` when generating the journal ID didn't seem right because I want it to be reproducible for the same process.

Reply:
Ah, I see. A comment describing what you just said would be useful.

Good to see there's an easy way to make things unique.

Reply:
Made journal ID unique by including the timestamp.